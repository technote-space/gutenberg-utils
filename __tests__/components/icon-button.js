/* eslint-disable no-magic-numbers */
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import IconButton from '../../src/components/icon-button';
import IconButtonImplement from '../../src/components/icon-button/implement';

describe('IconButton', () => {
	it('should render IconButton', () => {
		const wrapper = render(
			<div>
				<IconButton
					icon={'http://example.com'}
					className={'test1'}
				/>
				<IconButton
					icon={'data:image/png;base64,1234567890'}
					className={'test2'}
				/>
				<IconButton
					icon={'dashicons-menu'}
					className={'test3'}
				/>
			</div>,
		);

		expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();

		expect(wrapper.find('button.has-icon')).toHaveLength(3);
		expect(wrapper.find('.test1')).toHaveLength(1);
		expect(wrapper.find('.test2')).toHaveLength(1);
		expect(wrapper.find('.test3')).toHaveLength(1);
	});
});

describe('IconButtonImplement', () => {
	it('should render IconButtonImplement', () => {
		const wrapper = render(
			<div>
				<IconButtonImplement
					icon={'http://example.com'}
					className={'test1'}
				/>
				<IconButtonImplement
					icon={'data:image/png;base64,1234567890'}
					className={'test2'}
				/>
				<IconButtonImplement
					icon={'dashicons-menu'}
					className={'test3'}
					tooltip={'tooltip '}
				/>
			</div>,
		);

		expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();

		expect(wrapper.find('button.has-icon')).toHaveLength(3);
		expect(wrapper.find('button[aria-label]')).toHaveLength(1);
		expect(wrapper.find('.test1')).toHaveLength(1);
		expect(wrapper.find('.test2')).toHaveLength(1);
		expect(wrapper.find('.test3')).toHaveLength(1);
	});
});
