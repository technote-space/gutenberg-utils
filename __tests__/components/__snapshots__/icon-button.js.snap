// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render IconButton 1`] = `
<div>
  <button
    class="components-button test1 has-icon"
    type="button"
  />
  <button
    class="components-button test2 has-icon"
    type="button"
  />
  <button
    class="components-button test3 has-icon"
    type="button"
  />
</div>
`;

exports[`IconButtonImplement should render IconButtonImplement 1`] = `
<div>
  <button
    class="components-button test1 has-icon"
    type="button"
  />
  <button
    class="components-button test2 has-icon"
    type="button"
  />
  <button
    aria-label="tooltip "
    class="components-button test3 has-icon"
    type="button"
  />
</div>
`;
